File: Observer/typescript/src/CurrentConditionsDisplay.ts
import IDisplayElement from "./IDisplayElement";
import IObserver from "./IObserver";
import ISubject from "./ISubject";

export class CurrentConditionsDisplay implements IDisplayElement, IObserver {
  private temperature = 0.0;
  private humidity = 0.0;
  private pressure = 0.0;
  private _weatherData: ISubject;

  constructor(weatherData: ISubject) {
    this._weatherData = weatherData;
    this._weatherData.registerObserver(this);
  }

  display() {
    console.log(`Current conditions: ${
      this.temperature} F degrees, ${
      this.humidity}% humidity and ${
      this.pressure} pressure.`);
  }

  update(temperature: number, humidity: number, pressure: number) {
    this.temperature = temperature;
    this.humidity = humidity;
    this.pressure = pressure;
    this.display();
  }
}

File: Observer/c#/CurrentConditionsDisplay.cs
namespace Observer
{
    public class CurrentConditionsDisplay: IDisplayElement, IObserver
    {
        private float temperature;
        private float humidity;
        private float pressure;
        private ISubject _weatherData;

        public CurrentConditionsDisplay(ISubject weatherData)
        {
            _weatherData = weatherData;
            _weatherData.registerObserver(this);
        }


        public void display()
        {
            System.Console.WriteLine($@"Current conditions: {
              temperature } F degrees, {
              humidity } % humidty and {
              pressure } in pressure.");
        }

        public void update(float temperature, float humidity, float pressure)
        {
            this.temperature = temperature;
            this.humidity = humidity;
            this.pressure = pressure;
            display();
        }
    }
}